### PR TITLE
Upgrade borsh to 1.0

### PR DESCRIPTION
**Pull request by ruseinov**
Work on https://github.com/near/near-account-id/issues/5.

### CODE REVIEW

**Review by frol — APPROVED** [review]
Looks good to me 👍

**Comment by ruseinov** [comment]
@frol I have bumped the minimum supported Rust version as borsh is currently a default feature. If we desire to do something else - let's coordinate.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,4 +42,15 @@ jobs:
     - name: Install clippy
       run: rustup component add clippy
     - name: Run clippy
-      run: cargo clippy --all-features
+      run: cargo clippy
+  clippy-borsh:
```
Comment (frol) [inline .github/workflows/ci.yml:46]: Let’s run clippy only for all features enabled to save CI resources

File: Cargo.toml [hunk 2]
```diff
@@ -9,11 +9,12 @@ license = "MIT OR Apache-2.0"
 publish = true
 
 [features]
-default = ["borsh", "serde"]
+default = ["serde"]
+borsh = ["dep:borsh"]
```
[the remark this answers is not in the document]
Comment (ruseinov) [reply Cargo.toml]: You are right!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -18,9 +18,20 @@ jobs:
     - uses: actions/checkout@v3
     - uses: Swatinem/rust-cache@v1
     - name: Build
-      run:  cargo build --all-features --verbose --locked
+      run:  cargo build --verbose --locked
     - name: Run tests
-      run: cargo test --all-features --workspace --verbose --locked
+      run: cargo test --workspace --verbose --locked
+  test-borsh:
```
[the remark this answers is not in the document]
Comment (ruseinov) [reply .github/workflows/ci.yml]: done, that was to preserve the older version of clippy, but to be honest that's not necessary. Fixed.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ruseinov) [reply .github/workflows/ci.yml]: int.checked_mul(percent as u128).unwrap_or_else(|| env::panic_str("royalty overflow")) / 100u128

**Comment by ruseinov** [comment]
All addressed!

**Review by frol — APPROVED** [review]

**Pull request merged**